plugins/destination/metrics.go

```diff
@@ -1,8 +1,27 @@
 package destination
 
+import (
+	"sync/atomic"
+)
+
 type Metrics struct {
 	// Errors number of errors / failed writes
 	Errors uint64
 	// Writes number of successful writes
 	Writes uint64
 }
+
+func (mx *Metrics) Get() Metrics {
+	return Metrics{
+		Errors: atomic.LoadUint64(&mx.Errors),
+		Writes: atomic.LoadUint64(&mx.Writes),
+	}
+}
+
+func (mx *Metrics) Failed(amount int) {
```

Comment:
seems strange why `errors` -> `failed` and `writes` -> `Wrote`. aslo what is the value of having those operations instead of just returning the struct. I don't see any race possible here in the copy

Reply:
I wanted to expose atomic methods for the public fields, didn't find better names.
As for the atomic for the load part: same logic that was used for atomic store applies.